addons/mail/security/mail_security.xml

```diff
@@ -71,4 +71,15 @@
             <field name="perm_unlink" eval="False"/>
         </record>
 
+        <record id="mail_user_settings_rule" model="ir.rule">
+            <field name="name">mail.user.settings: write its own entries</field>
+            <field name="model_id" ref="model_mail_user_settings"/>
+            <field name="groups" eval="[Command.link(ref('base.group_user'))]"/>
+            <field name="domain_force">[('user_id', '=', user.id)]</field>
+            <field name="perm_read" eval="False"/>
+            <field name="perm_write" eval="True"/>
+            <field name="perm_create" eval="False"/>
+            <field name="perm_unlink" eval="False"/>
+        </record>
+
```

Comment:
Not only write should be limited to own user, but all other operations also.

Reply:
fixed